### DISCUSSION TITLE
find text in node text/details/notice with formula created text

### DISCUSSION THREAD

**Discussion by KB168** [freeplane/freeplane#1567] [Help]
Hi community,
I tried to use text search filter for concated informations in a node
with the following example (SeachString is a defined attribute for identifying special children)
mynode.text: "= parent['SeachString']"
mynode.detail: "= parent['SeachString']"
mynode.notice: "= parent['SeachString']"
results in: only the first (mynode.text) is found in the build in search engine and the others are not.
Also the Window "Search and Replace" shows the resolution of mynode.text, the others only the formula itself (=parent...)!
My Question: Is there a solution using formula with details/notice to find textz without using groovy or further solutions?

**Comment by EdoFro** [comment]
> My Question: Is there a solution using formula with details/notice to find textz without using groovy or further solutions?

I think not.

I made a test in one of my maps and used the formula `= 3 + 5` as node's text, details and note for different nodes.

then I selected the option "Script filter" in the search toolbar and used following script:

```groovy
def searched = '8'

def n = node

def vals = []

vals 
    << n.displayedText
    << n.plainText
    << n.details?.plain
    << n.details?.string
    << n.note?.plain
    << n.note?.string
    
return vals.any{v -> htmlUtils.htmlToPlain(v)?.contains(searched) }

```

and worked the way you asked for.

I hope it helps,

edo

**Comment by KB168** [comment]
Hello edo,
many thanks for your work and answer!
Helps, yes and no, I hoped for a solution with formula but your answer showed me to think about other solutions.
